### PR TITLE
Add components

### PR DESCRIPTION
**Pull request by sainirmayi**
## Describe your changes

Move openproblems-v2 spatial decomposition task to separate repo.

## Checklist before requesting a review
- [x] I have performed a self-review of my code

- Check the correct box. Does this PR contain:
  - [ ] Breaking changes
  - [ ] New functionality (new method, new metric, ...)
  - [ ] Major changes
  - [ ] Minor changes
  - [ ] Bug fixes

- [ ] Proposed changes are described in the CHANGELOG.md

- [ ] CI Tests succeed and look good!

### CODE REVIEW

File: src/api/comp_control_method.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      direction: output
+      required: true
+  test_resources:
+    - type: python_script
+      path: /src/common/component_tests/check_method_config.py
+    - type: python_script
+      path: /src/common/component_tests/run_and_check_output.py
+    - path: /resources_test/spatial_decomposition/cxg_mouse_pancreas_atlas
+      dest: resources_test/spatial_decomposition/cxg_mouse_pancreas_atlas
+    - path: /src/common/library.bib
```
Comment (KaiWaldrant) [inline src/api/comp_control_method.yaml:38]:
```suggestion
    - path: /common/library.bib
```

File: src/api/comp_method.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      direction: output
+      required: true
+  test_resources:
+    - type: python_script
+      path: /src/common/component_tests/check_method_config.py
+    - type: python_script
+      path: /src/common/component_tests/run_and_check_output.py
+    - path: /resources_test/spatial_decomposition/cxg_mouse_pancreas_atlas
+      dest: resources_test/spatial_decomposition/cxg_mouse_pancreas_atlas
+    - path: /src/common/library.bib
```
Comment (KaiWaldrant) [inline src/api/comp_method.yaml:29]:
```suggestion
    - path: /common/library.bib
```

File: src/api/comp_metric.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      direction: output
+      required: true
+  test_resources:
+    - type: python_script
+      path: /src/common/component_tests/check_metric_config.py
+    - type: python_script
+      path: /src/common/component_tests/run_and_check_output.py
+    - path: /resources_test/spatial_decomposition/cxg_mouse_pancreas_atlas
+      dest: resources_test/spatial_decomposition/cxg_mouse_pancreas_atlas
+    - path: /src/common/library.bib
```
Comment (KaiWaldrant) [inline src/api/comp_metric.yaml:30]:
```suggestion
    - path: /common/library.bib
```

File: .github/workflows/test.yml
Comment (KaiWaldrant) [inline .github/workflows/test.yml:1]: Maybe add this in a separate PR and merge it. Then merge main in this PR en let it rerun the CI.

Comment (sainirmayi) [reply .github/workflows/test.yml:1]: Looks like I just needed to add this to the checkout action in the `viash_test` job too.

**Review by rcannood — APPROVED** [review]

**Pull request merged**